### PR TITLE
Improve design of collection management classes

### PR DESCRIPTION
**Pull request by gap777**
This PR builds upon #17, which itself introduces a bug fix for using the custom options in page 2..N queries by the collection management classes.

Two different abstractions seemed to present themselves: HarvestRecord (e.g User, Project, Task) and HarvestRecordCollection (e.g. Users, Projects, Tasks). There are some similarities (their attributes are passed in as a hash), but there are many differences between the two concepts:

* HarvestRecord - #save, #create, #update; and unique attributes from one record type to another 
* HarvestRecordCollection - Harvesting::Enumerable, #page, #size,  #fetch_next_page, common attributes across all record collections

This change reduces the duplication between the concrete record collection classes (Users, Projects, Tasks, TimeEntries) while maintaining the behavior of #17.

### CODE REVIEW

**Comment by gap777** [comment]
@etagwerker @benphelps @mscottford can we get some response to this? Seems like a few PRs have stalled.

File: lib/harvesting/models/harvest_record_collection.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module Harvesting
+  module Models
+    class HarvestRecordCollection < Base
```
Comment (etagwerker) [inline lib/harvesting/models/harvest_record_collection.rb]: @gap777 I think this is a good idea.

File: lib/harvesting/models/harvest_record.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Harvesting
+  module Models
+    class HarvestRecord < Base
```
Comment (etagwerker) [inline lib/harvesting/models/harvest_record.rb]: @gap777 I think this is a good idea.

**Comment by gap777** [comment]
@etagwerker ready for merge, I think.

**Pull request merged**